### ISSUE TITLE
`withSizeLimit` behaves differently for multipart form uploads between 10.1.4 and 10.1.5

### ISSUE THREAD

**Issue by GlassAndOneHalf** [akka/akka-http#2214]
I have been upgrading my codebase to use akka-http `10.1.5` from `10.1.4` and I noticed that the `withSizeLimit` directive now seems to count the metadata of multipart form uploads towards the limit.

I have tried to produce a minimal set of tests that demonstrate this. 

```scala
class WithSizeLimitSpec extends WordSpec with Matchers with ScalatestRouteTest {

  val sizeLimit = 500

  val multipartUploadRoute: Route =
    (put & withSizeLimit(sizeLimit) & fileUpload("file")) { case (metadata, file) =>
      complete(StatusCodes.OK)
    }

  "A multipart upload with a limit" should {
    "accept uploads below the limit" in {
      val source = HttpEntity.limitableByteSource(Source.single(ByteString("0" * (sizeLimit - 1))))
      val data = Multipart.FormData(
        Multipart.FormData.BodyPart(
          "file",
          HttpEntity(MediaTypes.`image/png`, sizeLimit - 1, source),
          Map("filename" -> "test-file.png")
        )
      )

      Put("/", data) ~> multipartUploadRoute ~> check {
        status shouldEqual StatusCodes.OK
      }
    }

    "accept uploads equal to the limit" in {
      val source = HttpEntity.limitableByteSource(Source.single(ByteString("0" * sizeLimit)))
      val data = Multipart.FormData(
        Multipart.FormData.BodyPart(
          "file",
          HttpEntity(MediaTypes.`image/png`, sizeLimit, source),
          Map("filename" -> "test-file.png")
        )
      )

      Put("/", data) ~> multipartUploadRoute ~> check {
        status shouldEqual StatusCodes.OK
      }
    }

    "reject uploads above the limit" in {
      val source = HttpEntity.limitableByteSource(Source.single(ByteString("0" * (sizeLimit + 1))))
      val data = Multipart.FormData(
        Multipart.FormData.BodyPart(
          "file",
          HttpEntity(MediaTypes.`image/png`, sizeLimit + 1, source),
          Map("filename" -> "test-file.png")
        )
      )

      Put("/", data) ~> multipartUploadRoute ~> check {
        // EntityStreamSizeException => InternalServerError
        status shouldEqual StatusCodes.InternalServerError
      }
    }
  }
}
```

When these tests are run with akka-http `10.1.4` - they all pass. When run with `10.1.5`, the first two fail.

I am running these tests using the default akka-http configuration with Scala `2.12.6`, SBT `1.2.3` and the following dependencies:
```sbt
libraryDependencies ++= Seq(
  "com.typesafe.akka" %% "akka-http" % "10.1.5" % Test,
  "com.typesafe.akka" %% "akka-http-testkit" % "10.1.5" % Test,
  "org.scalatest" %% "scalatest" % "3.0.4" % Test
)
```

**Comment by jrudolph** [comment]
Hi @GlassAndOneHalf, thanks for the report.

What we fixed in 10.1.5 is that `withSizeLimit` will work regardless of whether you used `limitableByteSource` or not. In any case, what you see in 10.1.5 is the intended behavior: `withSizeLimit` should always apply to the full entity.

I haven't checked but I suspect that for `multipart/formdata` it actually worked the same also in 10.1.4 *on real servers* but not in tests, because in the testkit, adding the `limitableByteSource` was forgotten as well (which doesn't matter any more with 10.1.5).

**Comment by GlassAndOneHalf** [comment]
Hey @jrudolph, thanks for the quick response, I suspected that was the case but just wanted to confirm! 👍
